assets/js/modules/analytics-4/datastore/report.js

```diff
@@ -370,6 +370,34 @@ const baseSelectors = {
 
 		return isZeroReport( report );
 	} ),
+
+	/**
+	 * Gets a given report for each of the provided audiences.
+	 *
+	 * TODO: This will be refactored to use pivot reports in #8484.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Array.<Object>} An array of report response objects for each provided audience.
+	 */
+	getReportForAllAudiences: createRegistrySelector(
+		( select ) =>
+			( state, { options, audiences } ) => {
```

Comment:
As mentioned in a comment on the `AudienceTiles` component, this will need an update to accept audience names rather than objects, e.g.:

```suggestion
			( state, { options, audienceResourceNames } ) => {
```